### ISSUE TITLE
Modal dialogs turns UECIDE unresponsive to keyboard/mouse events

### ISSUE THREAD

**Issue by SantoRemedio** [UECIDE/UECIDE#124]
**Describe the bug**
Some combination of user actions with some dialogs (Preferences, Plugin Manager) make the app unresponsive to keyboard/mouse events. The only way out is to kill the app.

**To Reproduce**
1. Open UECIDE IDE.
2. Click on 'File - Preferences'.
3. Click on any part of the main UECIDE window underneath (to cover the dialog with the main window).
4. Presto, keyboard/mouse events not longer works for the app.

**Expected behavior**
Keyboard/mouse events must always work.

**Desktop (please complete the following information):**
 - OS: Linux Mint Tara
 - Uecide 0.10.4

**Other dialogs affected**
- About
- Sketch properties

**Comment by majenkotech** [comment]
> Click on any part of the main UECIDE window underneath (to cover the dialog with the main window).

I'm not sure I follow. That is not possible. The dialogs are application modal and block everything. You can't interact with the main app window while an application modal dialog is open. If the main window can become placed in front of the dialog then that is down to a fault with your window manager (or its interaction with Java).

**Comment by majenkotech** [comment]
Which WM are you using with Tara? Cinnamon, MATE or Xfce?

**Comment by majenkotech** [comment]
That link is broken.

Opening a modal dialog *will* block all input to the main app. That is what a modal dialog *does*.  It should be impossible to get the main app window to end up in front of the modal dialog window. If it can happen then it's not UECIDE that is letting it happen. 

I am downloading Tara MATE now to try it out.

**Comment by SantoRemedio** [comment]
Duh! Privacy settings defaults to private.

Try it now: https://youtu.be/_I_rGpMMXqI

[Window Manager Mate 1.20.1 (stock installation).]

**Comment by majenkotech** [comment]
That certainly should not be happening. And it's not something I have seen happen, and can't replicate on any other system.  Mint has nearly downloaded and I will set up a VM to test it out.  

There should be no way at all that the window manager should allow the dialog to get behind its parent window.

**Comment by majenkotech** [comment]
Ok, I replicate it on Mint.  It looks like something that it doesn't like with the custom window rendering of JTattoo.  Nothing I can do about that, it's not my code (and I have no clue how it works).  If you change the window style to "Gnome" in the theme preferences the problem rectifies itself.

To do it outside of UECIDE you can edit ~/.uecide/preferences.txt and look for `theme.laf=...` and set it to `gnome`.

**Comment by majenkotech** [comment]
Right, yes it's the custom decorator.  I did have code in there to allow it to be disabled, but for some reason it wasn't working. (Actually that "some" reason was that it was being run too late to take effect).  I have fixed that code now and it honours the setting.  It should default to disabled which will fix the problem, but the theme won't look *quite* as good as it would since it will now have the system default window decorations. 

But at least the choice will be yours: custom window decorations and possible dodgy modality management, or system window decorations and fixed modality management.
